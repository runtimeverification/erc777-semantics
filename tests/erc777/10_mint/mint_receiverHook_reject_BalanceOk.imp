test.preamble();

// operatorMint() through default operator (unless otherwise specified), operatorMint() from caller.
// receiver hook, always rejecting, minting new balance

test.registerTokensReceivedHook(5, 3); // Hook address 3 is rejecting.
test.makeCaller(5);
test.operatorMint(5, 1000, "", ""); //from caller

test.makeCaller(8);
test.operatorMint(5, 1000, "", ""); //through default operator
