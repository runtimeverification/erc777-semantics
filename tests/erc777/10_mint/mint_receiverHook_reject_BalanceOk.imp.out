Registering receiver 5 with hook 3
Changing the caller to 5
5 is minting 1000 for 5 (data:""),  (operatorData:"")
tokensReceived: Hook address = 3, Operator = 5, From = 0, To = 5, Value = 1000, Data = , OperatorData = 
tokensReceived: sender balance: 0
tokensReceived: receiver balance: 1050
Reverting...
REVERT
Balance of 5 is 50
Total supply is 550
Changing the caller to 8
8 is minting 1000 for 5 (data:""),  (operatorData:"")
tokensReceived: Hook address = 3, Operator = 8, From = 0, To = 5, Value = 1000, Data = , OperatorData = 
tokensReceived: sender balance: 0
tokensReceived: receiver balance: 1050
Reverting...
REVERT
Balance of 5 is 50
Total supply is 550
